unstructured_ingest/v2/processes/connectors/kafka.py

```diff
@@ -0,0 +1,324 @@
+    def kafka_consumer(self):
+        if self._consumer is None:
+            self._consumer = self.create_consumer()
+        return self._consumer
+
+    @requires_dependencies(["confluent_kafka"], extras="kafka")
+    def create_consumer(self) -> "Consumer":
+        from confluent_kafka import Consumer
+
+        is_confluent = self.connection_config.confluent
+        bootstrap = self.connection_config.bootstrap_server
+        port = self.connection_config.port
+
+        conf = {
+            "bootstrap.servers": f"{bootstrap}:{port}",
+            "client.id": socket.gethostname(),
+            "group.id": "your_group_id",
+            "enable.auto.commit": "false",
+            "auto.offset.reset": "earliest",
+            "message.max.bytes": 10485760,
+        }
+
+        if is_confluent:
+            kafka_api_key = self.connector_config.access_config.kafka_api_key
+            secret = self.connector_config.access_config.secret
+            conf["sasl.mechanism"] = "PLAIN"
+            conf["security.protocol"] = "SASL_SSL"
+            conf["sasl.username"] = kafka_api_key
+            conf["sasl.password"] = secret
+
+        consumer = Consumer(conf)
+        logger.debug(f"kafka consumer connected to bootstrap: {bootstrap}")
+        return consumer
+
+
+    def precheck(self):
+        try:
+            self.kafka_consumer
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise SourceConnectionError(f"failed to validate connection: {e}")
+
+class KafkaDownloaderConfig(DownloaderConfig):
+    pass
+
+@dataclass
+class KafkaDownloader(Downloader):
+    connection_config = KafkaConnectionConfig
+    download_config = KafkaDownloaderConfig = field(
+        default_factory = lambda: KafkaDownloaderConfig
+    )
+    connector_type: str = "kafka"
+    raw_content: str
+    raw_filename: str
+    registry_name: str = "kafka"
+
+    def _tmp_download_file(self):
+        topic_file = self.connection_config.topic + "-" + self.raw_filename
+        return Path(self.download_dir) / topic_file
+
+    @property
+    def version(self) -> t.Optional[str]:
+        return None
+
+    @property
+    def source_url(self) -> t.Optional[str]:
+        return None
+
+    @property
+    def filename(self):
+        """The filename of the file created"""
+        return self._tmp_download_file()
+
+    def _create_full_tmp_dir_path(self):
+        self._tmp_download_file().parent.mkdir(parents=True, exist_ok=True)
+
+    @SourceConnectionError.wrap
+    def _get_file(self):
+        self._create_full_tmp_dir_path()
+
+        pdf_data = base64.b64decode(self.raw_content)
+
+        with open(self.filename, "wb") as file:
+            file.write(pdf_data)
+
+    @SourceConnectionError.wrap
+    def run(self, file_data: FileData, **kwargs: t.Any) -> download_responses:
+        self._get_file()
+
+        return DownloadResponse(file_data=file_data, path=Path(self.filename))
```

Comment:
Should wrap this in a `generate_download_response` call. 
```suggestion
        return self.generate_download_response(file_data=file_data, download_path=download_path)
```